### ISSUE TITLE
Game crash on movement with soundpacks with movement sounds.

### ISSUE THREAD

**Issue by Varenik1226** [CleverRaven/Cataclysm-DDA#48192]
# Describe the bug
Not played since Version 0.E-7938-g8736e6b (tiles) - Build 11203 (Win x64).
I have installed latest experimental version (0.E-10149-g894002b (tiles) - build 11539 (Win x64)) in new folder with latest Otopack soundpack. And when I started a new game, after a step in any direction, the game crashes. Same issue with @'s Soundpack. After that, I tried to delete pieces of code in the soundpack file "player.json" with id "plmove". And the game stopped crashing.
I still have a folder with build 11203. On that version everything works fine.

# Steps To Reproduce

1. Install CDDA
2. Install soundpack with movement sounds and choose it in options.
3. Start new game
4. Make a step in any direction.
5. After the crash, enjoy your desktop wallpaper.

# Expected behavior
Take more than one step in the game with walking sounds 

# Screenshots

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 0.E-10149-g894002b [64-bit]
- Graphics Version: Tiles
- Game Language: Русский [ru]
- Mods loaded: [
    Тёмные Дни Впереди [dda],
    Отключить потребности у NPC [no_npc_food]
]

# Additional context

There is no crash.log file
[debug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6189410/debug.zip)
It happens with all new worlds and characters, but maybe you need it.
[Save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6189424/Save.zip)

**Comment by StefanMiletic994** [comment]
I've got this problem too I use RFF sounds, tried creating new character and still crashes after I updated a few moments ago.

**Comment by actual-nh** [comment]
Does it crash if you remove the soundpack from your configuration?

**Comment by chrispikula** [comment]
So I've got a repeatable instance of this, whenever I go up the stairs out of the basement, instant crash.

Would it have anything to do with getting rid of opengl2 in the option file?

[Grantsboro.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6190962/Grantsboro.zip)

[options.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/6190979/options.zip)

Game version:
0.E-10149-g894002b [64-bit], Tiles, Build 11539

**Comment by coyo7e** [comment]
this hit me too with RRFSounds pack as son as I updated, with a loaded and a new character I made as a test. reverting to previous version seems to have addressed it for the time being.

**Comment by actual-nh** [comment]
What was the most recent version not giving a crash (ideally, with build number)?

**Comment by Zireael07** [comment]
@coyo7e: What is 'previous version' for you?

**Comment by chrispikula** [comment]
Reverting to 0.E-10134-g9b256f6 (tiles) build 11537 fixed the issue for me as well.

I only reverted options.json, VERSION.txt, cataclysm-tiles.exe, and json_formatter.exe.  Everything else was left the same.  This was done via WinMerge, a file/line-by-line diff utility.

**Comment by coyo7e** [comment]
> 
> 
> @coyo7e: What is 'previous version' for you?

- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-10114-gd55d4a1 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Alternative Map Key [alt_map_key],
    No Rail Stations [No_Rail_Stations],
    No Fungal Monsters [No_Fungi],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Fuji's Military Profession Pack [fuji_mpp],
    Blaze Industries [blazeindustries]
]

**Comment by actual-nh** [comment]
Is a 11538 build for Windows x64 available?

**Comment by chrispikula** [comment]
It had a name that the launcher didn't like, let me check via direct download.

**Comment by actual-nh** [comment]
> Edit: Also, did the extra option in options.json build 11539 vs 11537 bypass string freeze?

It was changing a default, not putting in an extra option (or at least should not have been).

**Comment by BrettDong** [comment]
> Is a 11538 build for Windows x64 available?

Nope, 11538 Windows x64 build timed out, but Windows x86 is available.

**Comment by actual-nh** [comment]
> > Is a 11538 build for Windows x64 available?
> 
> Nope, 11538 Windows x64 build timed out, but Windows x86 is available.

The question then is whether 11539 errors on a Windows x86 build.

**Comment by chrispikula** [comment]
> > Edit: Also, did the extra option in options.json build 11539 vs 11537 bypass string freeze?
> It was changing a default, not putting in an extra option (or at least should not have been).
Ah, my bad.  I must not have merged the options.json update in a while, turns out it was the removal of an option that I got mixed up with, with all the versioning back and forth.
Turns out options.json removed the following between 11537 and 11539:
`"Cuteclysm(Alpha), " line 106
"opengles2, " line 125
"{ "info": "Use render batching for 2D render API to make it more efficient.  Requires restart.", "default": "Default: True", "name": "RENDER_BATCHING", "value": "true" }, " line 126
`

**Comment by actual-nh** [comment]
> > > Edit: Also, did the extra option in options.json build 11539 vs 11537 bypass string freeze?
> 
> > It was changing a default, not putting in an extra option (or at least should not have been).
> 
> Ah, my bad. Turns out it was the removal of an option that I got mixed up with, with all the versioning back and forth.
> Turns out options.json removed the following between 11537 and 11539:
> `"Cuteclysm(Alpha), " line 106`
> `"opengles2, " line 125`
> `"{ "info": "Use render batching for 2D render API to make it more efficient. Requires restart.", "default": "Default: True", "name": "RENDER_BATCHING", "value": "true" }, " line 126`

Blink. Options.json - how did the version change alter a JSON file that's user-generated? (Admittedly, I'm on a mac and using the USE_HOME_DIR compilation option, so it may be different for me.)

**Comment by Varenik1226** [comment]
> 
> 
> Does it crash if you remove the soundpack from your configuration?

After removing soundpack, game works fine.

**Comment by Varenik1226** [comment]
If necessary, I can provide remote access to my pc for tests.

**Comment by actual-nh** [comment]
> If necessary, I can provide remote access to my pc for tests.

Enough developers use windows x64 that that should not be needed, I think (and would be better discussed someplace where private messaging is possible!); thank you, though!

**Comment by ZhilkinSerg** [comment]
`mxe` (and linked libs, like various `SDL2` libs) version was bumped on Jenkins before build #11539 was triggered.

**Comment by ZhilkinSerg** [comment]
> Edit: Also, did the extra option in options.json build 11539 vs 11537 bypass string freeze?

No, string was there for a long time - option was just hidden in old SDL2 version we were using.

**Comment by Varenik1226** [comment]
> Enough developers use windows x64 that that should not be needed, I think (and would be better discussed someplace where private messaging is possible!); thank you, though!

of course I will not provide my data here =)

**Comment by Varenik1226** [comment]
For me? the game crashes with any default tilesets. 0.E-10149-g894002b (tiles) - build 11539 (Win x64)

**Comment by actual-nh** [comment]
> of course I will not provide my data here =)

I didn't think so, but habit from being a security admin a while back...

> For me, the game crashes with any default tilesets. 0.E-10149-g894002b (tiles) - build 11539 (Win x64)

OK. So looks like the "Cuteclysm" change that @chrispikula noted is not responsible; not surprised, but something to check.

**Comment by anothersimulacrum** [comment]
Someone should test porting https://github.com/cataclysmbnteam/Cataclysm-BN/pull/355

**Comment by ProvoGo** [comment]
having the same problem using @'s soundpack build 11359 (win x64)

**Comment by RoyBerube** [comment]
On Windows 10 and using otopack for sound.

-  If I build the game on the latest master and using Vcpkg last compiled on December 18, no crash. 
- Using build 11540 - crash when taking a step.

**Comment by actual-nh** [comment]
Ping: @narc0tiq?

**Comment by BrettDong** [comment]
Reproducible using Windows x64 build 11539 with [Otopack soundpack](https://discourse.cataclysmdda.org/t/soundpack-otopack/23190), instant crash after moving two maptiles:
![image](https://user-images.githubusercontent.com/21075502/112251978-9ae6c300-8c32-11eb-8fa6-80d1818e2e5b.png)
Probably due to an incompatible combination of libraries on build server.

**Comment by BrettDong** [comment]
Stack trace also fails on build 11539 (Windows x64). I intentionally crashed the program via debug menu on both build 11539 (left) and build 11535 (right). The stack trace given in build 11539 is apparently nonsense:
![image](https://user-images.githubusercontent.com/21075502/112255642-fb78fe80-8c38-11eb-9f6d-9047ce8225f8.png)

**Comment by Varenik1226** [comment]
Found strange thing.
Installed 11537 x64 build, everything works fine with sounpack.
Then 11539 x64 build. Even with basic soundpack game feel very laggy in comparsion with 11537....very, veryyy laggy (default render Direct3D). If i change render on any other, lags disappear, but soundpack problems still here =(.

**Comment by AMurkin** [comment]
That's #48179, probably. Fixed since build 11540.

**Comment by Theawesomeboophis** [comment]
Problem fixed for me on current version with Otopack. Can you test to see if the problems are still occurring for you on current version?

**Comment by RoyBerube** [comment]
Started working for me with version 11543 yesterday. I'm curious what fixed it.

**Comment by Theawesomeboophis** [comment]
> Started working for me with version 11543 yesterday. I'm curious what fixed it.

I think it may have something to do with the demangling PR, as the update that contained that is when it started working for me.

**Comment by actual-nh** [comment]
> > Started working for me with version 11543 yesterday. I'm curious what fixed it.
> 
> I think it may have something to do with the demangling PR, as the update that contained that is when it started working for me.

That was in with 11544, as far as I can tell.

**Comment by Varenik1226** [comment]
For 11548 build (x64) on Direct3D render, the game still veeery laggy, checked in options. Game didn't crash on walking, but a strange metallic sound is added to the sounds of walking, as if something metallic hit on a mesh metal fence. (sry for google translate, I hope you understand what this is about).

**Comment by anothersimulacrum** [comment]
@ZhilkinSerg I think it may be worth leaving the sdl upgrade to after the release, where we have time to deal with these issues.

**Comment by ZhilkinSerg** [comment]
Please test performance in build 11550.

**Comment by Varenik1226** [comment]
11550 Win x64
Lags are still there =(
Strange metallic sounds when walking still there too.

**Comment by ZhilkinSerg** [comment]
Please try 11551 when it is out.

**Comment by Varenik1226** [comment]
> 
> 
> Please try 11551 when it is out.

How long will it take to compile approximately? It's just that it's 3:30 am already. If more than a half hour, then only tomorrow.

**Comment by ZhilkinSerg** [comment]
32 bit is out already, but there is no rush to test today.

**Comment by Varenik1226** [comment]
Here this sounds, build 11551 x86
https://user-images.githubusercontent.com/74045419/112876475-48654500-9111-11eb-8fd4-b57b76d8c415.mp4

**Comment by ZhilkinSerg** [comment]
> 
> 
> @ZhilkinSerg I think it may be worth leaving the sdl upgrade to after the release, where we have time to deal with these issues.

Yeah, I agree - we need to test new versions more during next experimental cycle.

**Comment by Varenik1226** [comment]
Win x64 build - 11566
Weird sound still there =(
Playing on 11537 =(

**Comment by actual-nh** [comment]
Is the sound problem still there, and is anyone else getting it?

**Comment by Theawesomeboophis** [comment]
I'm not getting it, can you (the creator of the issue) reinstall the
soundpack and see if the problem persists?

**Comment by Varenik1226** [comment]
> 
> 
> Is the sound problem still there, @Varenik1226? Assuming that it is:
> 
>     * Is it present with other soundpacks?
> 
>     * Could you change the title of the issue to reference metallic sounds when moving, to help check if anyone else is hearing it?

Found what was the problem, I have a "Creative SB AUDIGY RX 7.1" sound card installed. In the Windows sound settings, I have 5.1 sound. Metal sounds disappear when I set it to 2/2.1 sound. With 11537 build everything was fine with 5.1.

Just checked it with build11576:
5.1 sound - metal still here
2/2.1 sound - normal sounds.

I reinstalled the driver, tried 2 different versions, the result did not change.

**Comment by Theawesomeboophis** [comment]
Are there any updates available for your sound driver, if so, can you
update and retest?

**Comment by Varenik1226** [comment]
Tested 2 versions: 2015 and 2018 (latest). For now using 2018

**Comment by Theawesomeboophis** [comment]
Is it working on 2018?

**Comment by Varenik1226** [comment]
Only in 2/2.1 =)

**Comment by Theawesomeboophis** [comment]
Do you have any other drivers installed and if so, does it work on them?

**Comment by Varenik1226** [comment]
nope, Realtek soundcard disabled in BIOS.

**Comment by Varenik1226** [comment]
10 mins, i will turn it on and check

**Comment by Theawesomeboophis** [comment]
Can it be reenabled to test, or is it permanently disabled?

**Comment by Theawesomeboophis** [comment]
Ok, thank you.

**Comment by Varenik1226** [comment]
With Realtek everything same:
5.1 - metal
2/2.1 - norm

**Comment by Theawesomeboophis** [comment]
Does reinstalling the soundpack have any effect?

**Comment by Varenik1226** [comment]
nope, checked with Otopack and @'s soundpacks, on fresh installed CDDA.

**Comment by Varenik1226** [comment]
Seems like sound problem exist only with movement sound

**Comment by Theawesomeboophis** [comment]
Hmm, I'm not sure how to fix this then, hopefully someone finds a solution.

**Comment by Varenik1226** [comment]
For now i just set it to 2/2.1. And back again to 5.1 after playing.

**Comment by Alexalmighty502** [comment]
While I posted this in #49642 ill post it here. Here a small vid of the most extreme audio issue relating to surround sound https://www.youtube.com/watch?v=OHZxzCxulaI

**Comment by EirenexTheDragon** [comment]
> 
> 
> Seems like sound problem exist only with movement sound

I have endured the sound glitch for a long time and what i could make out what makes weird sounds is:
-Walking on Street/hard floor in general with footgear
- Hitting zombies, doesnt matter if fist or spear
- Vehicle sounds, even the sound of changing gear has that sound!
- very rarely also a metallic sound when walking on dirt

Changing Soundpacks, even to older ones, didnt seemed to help sadly. 
i hope this helps to locate the issue. 
i am using windows 10 and windows sound is set 7.1, i cant force it to 2.1 sadly, probably because of the headset i am using
headphone i am using is Logitech G432 which comes with its own software too, sadly disabling surround in the program does not remove the metall sound either.

**Comment by olanti-p** [comment]
Possible fix: https://github.com/cataclysmbnteam/Cataclysm-BN/pull/863
@EirenexTheDragon confirmed it solved the issue for walking sounds in BN.
